app/models/flag.rb

```diff
@@ -1,2 +1,3 @@
 class Flag < ActiveRecord::Base
+  has_and_belongs_to_many :programs
```

Comment:
Declining to change because the link between programs and flags is not rich; there is nothing to be gained by has_many :through.